src/org/mozilla/javascript/NativeObject.java

```diff
@@ -507,6 +513,30 @@ public Object execIdCall(
                     Scriptable desc = obj.getOwnPropertyDescriptor(cx, nameArg);
                     return desc == null ? Undefined.instance : desc;
                 }
+            case ConstructorId_getOwnPropertyDescriptors:
```

Comment:
EcmaScript spec says the getOwnPropertyDescriptors impl should use the [CreateDataPropertyOrThrow](https://262.ecma-international.org/12.0/#sec-createdatapropertyorthrow) Abstract Object operation

We don't have such method yet in AbstractEcmaObjectOperations yet. Would it make sense to refactor things (assuming that the logic of that Abstract Operation is already covered elsewhere) into the appropriate method in AbstractEcmaObjectOperations?

Reply:
Perhaps `CreateDataPropertyOrThrow` looks like the same behavior as `defineOwnProperty(cx, id, desc, checkValid = true)`.
https://github.com/mozilla/rhino/blob/c275c35f160ef7b035002c67c3540ba317964351/src/org/mozilla/javascript/ScriptableObject.java#L1569-L1570

I could use `defineOwnProperty` here, but I couldn't think of a test case. And, I think we will need `FromPropertyDescriptor` too. If any test case is violated by this PR, I will be fixed it.